packages/react-devtools-shared/src/constants.js

```diff
@@ -39,6 +39,9 @@ export const SESSION_STORAGE_LAST_SELECTION_KEY =
 export const LOCAL_STORAGE_OPEN_IN_EDITOR_URL =
   'React::DevTools::openInEditorUrl';
 
+export const LOCAL_STORAGE_OPEN_IN_EDITOR_URL_PRESET =
+  'React::DevTools::openInEditorUrlPreset';
+
```

Comment:
I suggest removing this new config, then if the `React::DevTools::openInEditorUrl` equals to `'vscode://file/{path}:{line}'`, you select the `VSCode` option. That will be simpler.

Reply:
hi @Biki-das will you consider this?

Reply:
> hi @Biki-das will you consider this? also cc @hoxyq

I think this adds kinda wrong dependency, you should select the preset first and then based on its value, you either do nothing or update the link template.

For example, if I select `"custom"` and then put `"vscode://file/{path}:{line}"`, should we automatically switch to `VSCode` option and hide input?

Reply:
> > hi @Biki-das will you consider this? also cc @hoxyq
> 
> I think this adds kinda wrong dependency, you should select the preset first and then based on its value, you either do nothing or update the link template.
> 
> For example, if I select `"custom"` and then put `"vscode://file/{path}:{line}"`, should we automatically switch to `VSCode` option and hide input?

yup current implementation looks right i think @hoxyq

Reply:
> For example, if I select "custom" and then put "vscode://file/{path}:{line}", should we automatically switch to VSCode option and hide input?

Yes. I think this is good, but plz continue this PR if you don't agree with me!